### PR TITLE
glib: add 2.76.3 and 2.76.2, update build systems

### PR DESCRIPTION
**Pull request by michaelkuhn**
This also converts the package to the new way of handling multiple build systems.

### CODE REVIEW

File: var/spack/repos/builtin/packages/glib/package.py [hunk 1]
```diff
@@ -118,6 +119,12 @@ class Glib(Package):
...
 
+    build_system(
+        conditional("meson", when="@2.58:"),
+        conditional("autotools", when="@:2.57"),
+        default="meson",
+    )
+
     depends_on("meson@0.49.2:", when="@2.61.2:", type="build")
     depends_on("meson@0.48.0:", when="@2.58.0:", type="build")
     depends_on("ninja", when="@2.58.0:", type="build")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/glib/package.py]: Maybe wrap those in

```python
with when("build_system=meson"):
  ...
```

and drop the `@2.58.0:` lowerbound

Comment (michaelkuhn) [reply var/spack/repos/builtin/packages/glib/package.py]: Good point, done! I've also removed the ninja dependency added implicitly by `MesonPackage` and updated the required meson versions for the later releases.

**Comment by michaelkuhn** [comment]
I've pushed a new version that fixes the conflicts caused by #35450.

**Comment by michaelkuhn** [comment]
@alalazo @haampie ping

File: var/spack/repos/builtin/packages/glib/package.py [hunk 2]
```diff
@@ -180,6 +190,86 @@ def patch(self):
...
+            )
+            repl = "#!{0}".format(self.spec["python"].command.path)
+            files = ["glib-genmarshal", "glib-mkenums"]
+        else:
+            pattern = "^#! /usr/bin/perl"
+            repl = "#!{0}".format(self.spec["perl"].command.path)
+            files = ["glib-mkenums"]
+
+        files = [join_path(self.prefix.bin, file) for file in files]
+        filter_file(pattern, repl, *files, backup=False)
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/glib/package.py:253]: Functions decorated with `run_after` and `run_before` cannot stay in the `*Package` when the recipe supports multiple build-systems. I think these functions are currently never run.

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/glib/package.py (hunk 2 above)
Comment (michaelkuhn) [reply var/spack/repos/builtin/packages/glib/package.py:253]: Thanks for catching this! Is there a nicer way than simply copying all these methods into both `Builder` classes (which seems to work)?

Comment (alalazo) [reply var/spack/repos/builtin/packages/glib/package.py:253]: @michaelkuhn You can do it like it is done here https://github.com/spack/spack/pull/36825/files in the `netcdf-c` package. Insert the base builder last in the MRO and everything should be fine.

Comment (michaelkuhn) [reply var/spack/repos/builtin/packages/glib/package.py:253]: Thanks for the hint! That was easier than I thought. :smile: I've pushed a new version that should now work correctly.

**Review by alalazo — DISMISSED** [review]

**Comment by alalazo** [comment]
@spackbot fix style

**Review by alalazo — APPROVED** [review]

**Pull request merged**